### PR TITLE
refactor(metrics): use exponential_buckets

### PR DESCRIPTION
**Pull request by Sunt-ing**
## What's changed and what's your intention?
I refactored all Histogram buckets from `DEFAULT_BUCKETS+scale` or `manually setting every bucket` to exponential_buckets.

Reasons:
- Parameters in exponential_buckets are simple, straightforward, but powerful. In exponential_buckets, we can adjust three parameters: start, factor, bucket_count. 
  - Compare to using `DEFAULT_BUCKETS+scale`, it's more flexible. 
  - Compare to using `manually setting every bucket`, it's easier to tune or understand.
- TiKV also uses this method.

Currently, the parameters used in this PR are set according to the previous parameters and my experience.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
close https://github.com/singularity-data/risingwave/issues/1006

### CODE REVIEW

File: src/meta/src/rpc/metrics.rs [hunk 1]
```diff
@@ -67,20 +62,18 @@ pub struct MetaMetrics {
 impl MetaMetrics {
     pub fn new() -> Self {
         let registry = prometheus::Registry::new();
-        let buckets = DEFAULT_BUCKETS;
         let opts = histogram_opts!(
             "meta_grpc_duration_seconds",
             "gRPC latency of meta services",
-            buckets.iter().map(|x| *x * 0.1).collect_vec()
+            exponential_buckets(0.0001, 2.0, 16).unwrap() // max 6s
```
Comment (Little-Wallace) [inline src/meta/src/rpc/metrics.rs:68]: 6s may be too small for some heavy rpc

File: src/meta/src/rpc/metrics.rs [hunk 2]
```diff
@@ -67,20 +62,18 @@ pub struct MetaMetrics {
...
         let grpc_latency =
             register_histogram_vec_with_registry!(opts, &["path"], registry).unwrap();
 
-        let buckets = BARRIER_BUCKETS;
         let opts = histogram_opts!(
             "meta_barrier_duration_seconds",
-            "barrier latency ",
-            buckets.to_vec()
+            "barrier latency",
+            exponential_buckets(0.000005, 1.6, 36).unwrap() // max 115s
```
Comment (Little-Wallace) [inline src/meta/src/rpc/metrics.rs:76]: Is the start latency  5us? It seems too small..... Our barrier can not run so fast.....I suggest 100ms instead.

**Comment by Sunt-ing** [comment]
Changed. @Little-Wallace

File: src/meta/src/rpc/metrics.rs [hunk 3]
```diff
@@ -67,20 +62,18 @@ pub struct MetaMetrics {
...
         let grpc_latency =
             register_histogram_vec_with_registry!(opts, &["path"], registry).unwrap();
 
-        let buckets = BARRIER_BUCKETS;
         let opts = histogram_opts!(
             "meta_barrier_duration_seconds",
-            "barrier latency ",
-            buckets.to_vec()
+            "barrier latency",
+            exponential_buckets(0.1, 1.5, 16).unwrap() // max 65s
```
[the remark this answers is not in the document]
Comment (Sunt-ing) [reply src/meta/src/rpc/metrics.rs]: Fixed all similar miscalculations

File: src/storage/src/monitor/state_store_metrics.rs [hunk 4]
```diff
@@ -190,69 +157,56 @@ impl StateStoreMetrics {
...
 
-        let buckets = DEFAULT_BUCKETS
-            .map(|x| x * BATCH_WRITE_ADD_L0_LATENCT_SCALE)
-            .to_vec();
         let opts = histogram_opts!(
             "state_store_shared_buffer_to_l0_duration",
-            "Histogram of time spent from compacting shared buffer to remote storage.",
-            buckets
+            "Histogram of time spent from compacting shared buffer to remote storage",
+            exponential_buckets(0.0001, 2.0, 16).unwrap() // max 3s
```
Comment (Little-Wallace) [inline src/storage/src/monitor/state_store_metrics.rs:185]: 3s is too small.
we can start since 10ms and we can observe  640s at most

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Pull request merged**